lib/galaxy/tools/parameters/dynamic_options.py

```diff
@@ -555,14 +611,18 @@ def load_from_parameter(from_parameter, transform_lines=None):
             self.parse_column_definitions(elem)
             if data_file is not None:
                 data_file = data_file.strip()
-                if not os.path.isabs(data_file):
-                    full_path = os.path.join(self.tool_param.tool.app.config.tool_data_path, data_file)
+                full_path = os.path.join(self.tool_param.tool.app.config.tool_data_path, data_file)
+                full_path = os.path.normpath(full_path)
```

Comment:
Is the normpath call necessary here ? `safe_contains` should be smart enough to figure this out, and joining with `cwd` for non-absolute paths does not seem right (that seems to be what normpath does internally).

Reply:
I have undone these changes (they were not related or only related to the test).

Originally I wanted to test using options generated `from_file` (on which I would then apply the new filter). Problem was that `on_file` apparently only accepts absolute file paths which is difficult to use in tests where we have random directory names.